types/src/proof/accumulator/mod.rs

```diff
@@ -233,6 +233,14 @@ where
         self.root_hash
     }
 
+    pub fn version(&self) -> u64 {
+        if self.num_leaves() == 0 {
+            0
+        } else {
+            self.num_leaves() - 1
+        }
+    }
```

Comment:
I think this could be confusing.

Reply:
Should we instead return back None for num_leaves = 0 and num_leaves -1 otherwise? Or something else in mind? It feels weird that we have this being computed all over the place.

Reply:
I wanted to let genesis be version 1 for a looooooooooooog time.

Reply:
I actually asked this question yesterday lol, it is confusing